### ISSUE TITLE
It would be nice for `copy` to optionally show a progress bar during upload

### ISSUE THREAD

**Issue by edunham** [ansible/ansible#20579]
_From @edunham on February 29, 2016 21:53_

##### Issue Type:
- Feature Idea
##### Plugin Name:

copy
##### Ansible Version:

```
ansible 2.0.1.0
  config file = /home/.../ansible.cfg
  configured module search path = Default w/o overrides
```
##### Ansible Configuration:

```
$ cat ansible.cfg 
[defaults]
inventory = hosts
roles_path = roles/
```
##### Environment:

N/A, not platform-specific
##### Summary:

I have an ansible playbook which uses the `copy` module to upload a 3GB file to managed hosts. This takes a long time, and causes me to wonder whether it's still uploading or got stuck somewhere. It would be nice if the `copy` module had an option for showing a progress indicator while uploading.
##### Steps To Reproduce:

How the new feature might be used:

```
- name: Upload Xcode DMG to the host
  copy: 
      src={{ xcode_dmg }}
      dest=/Users/{{ ansible_ssh_user }}
      show_progress_bar=True
  when: xcode_dl_path.stat.exists == False
```
##### Expected Results:

It would be nice if the playbook output a progress indicator when the progress bar setting was enabled.
##### Actual Results:

I read the docs and asked on IRC and daemoniz confirmed that there is not a built-in feature to do this.

_Copied from original issue: ansible/ansible-modules-core#3149_

**Comment by edunham** [comment]
_From @ansibot on July 31, 2016 14:44_



@ansible ping, this issue is waiting for your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by edunham** [comment]
_From @ansibot on September 9, 2016 18:24_

@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by edunham** [comment]
_From @ansibot on October 6, 2016 14:57_

@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by edunham** [comment]
_From @ansibot on December 7, 2016 19:16_

This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by ryansb** [comment]
This would need a change in the way Ansible sends output from modules to be displayed, adding component label.

**Comment by kurtbomya** [comment]
I was just searching for a switch or something to enable this. I've OP's exact same issue. I'm copying a massive file and not sure if it's hung or still chugging. Can I go to lunch since it's working? ...or would that be irresponsible since this isn't doing anything? I can't tell.

EDIT: Temporary workaround. I actually found that ansible is copying to the remote machine's /tmp directory and I can keep refreshing the view to see the progress. (and verify that it's not hung)

**Comment by ansibot** [comment]
@edunham Greetings! Thanks for taking the time to open this issue. In order for the community to handle your issue effectively, we need a bit more information. 

Here are the items we could not find in your description:
- component name

Please set the description of this issue with this template:
https://raw.githubusercontent.com/ansible/ansible/devel/.github/ISSUE_TEMPLATE.md

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by mike-pt** [comment]
:+1: this would be a very nice addition

**Comment by hwangspringpath** [comment]
This is definitely a good feature to have.

**Comment by frippe75** [comment]
+1 from me

**Comment by Dan1elKa0** [comment]
+1

**Comment by ajslater** [comment]
Related issue: https://github.com/ansible/ansible/issues/3887

**Comment by maikelpenz** [comment]
+ 1

**Comment by maikelpenz** [comment]
It would be great to have it

**Comment by markovanderpuil** [comment]
+1 also +1 for rsync to have the same feature!

**Comment by maxkrylov** [comment]
+1 for packages modules (yum, apt) progress bar, very nice to have it)

**Comment by bcoca** [comment]
basically a subset of this https://github.com/ansible/proposals/issues/92, first it requires Ansible to have an 'updates from modules framework' and then each module/action needs to use it.

Closing in favor of the proposal.

**Comment by tarunmaini16** [comment]
+1, will be nice to have this feature!

**Comment by ryansb** [comment]
Locking this topic as conversation should move to the proposal

**Comment by abadger** [comment]
This is currently unachievable.  We'll look into it as part of introducing a framework that can periodically return information from modules rather than only on module completion.
